src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java

```diff
@@ -1712,6 +1712,116 @@ public void testGetSessionResultsForUser_orphanResponseInDB_shouldStillHandleCor
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
+
+    @Test
+    public void testResponseVisibleToStudentWhenReceiverExcludesSelf() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.STUDENTS_EXCLUDING_SELF)
+                .withGiverType(FeedbackParticipantType.STUDENTS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
+
+    @Test
+    public void testResponseVisibleToStudentInSameSection() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.STUDENTS_IN_SAME_SECTION)
+                .withGiverType(FeedbackParticipantType.STUDENTS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
+
+    @Test
+    public void testResponseVisibleToStudentWhenReceiverIsOwnTeamMember() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.OWN_TEAM_MEMBERS)
+                .withGiverType(FeedbackParticipantType.STUDENTS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
+
+    @Test
+    public void testResponseVisibleToStudentWhenReceiverIsGiver() {
+        List<FeedbackParticipantType> showResponseTo = new ArrayList<>();
+        showResponseTo.add(FeedbackParticipantType.RECEIVER);
+
+        FeedbackQuestionAttributes question = FeedbackQuestionAttributes.builder()
+                .withShowResponsesTo(showResponseTo)
+                .withRecipientType(FeedbackParticipantType.GIVER)
+                .withGiverType(FeedbackParticipantType.STUDENTS)
+                .build();
+        boolean response = FeedbackResponsesLogic.inst().isResponseOfFeedbackQuestionVisibleToStudent(question);
+        assertEquals(response, true);
+    }
```

Comment:
let's add a test case where isStudentRecipientType is false but question.getRecipientType().isTeam() is true, and another where isStudentRecipientType is true, but question.isResponseVisibleTo(FeedbackParticipantType.RECEIVER) is false